_includes/code/animation/radial_transition/main.dart

```diff
@@ -0,0 +1,180 @@
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              new SizedBox(
+                width: kMaxRadius * 2.0,
+                height: kMaxRadius * 2.0,
+                child: new Hero(
+                  createRectTween: _createRectTween,
+                  tag: imageName,
+                  child: new RadialExpansion(
+                    maxRadius: kMaxRadius,
+                    child: new Photo(
+                      photo: imageName,
+                      onTap: () {
+                        Navigator.of(context).pop();
+                      },
+                    ),
+                  ),
+                ),
+              ),
+              new Text(
+                description,
+                style: new TextStyle(fontWeight: FontWeight.bold),
+                textScaleFactor: 3.0,
+              ),
+              const SizedBox(height: 16.0),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildHero(BuildContext context, String imageName, String description) {
+    return new Container(
+      width: kMinRadius * 2.0,
+      height: kMinRadius * 2.0,
+      child: new Hero(
+        createRectTween: _createRectTween,
+        tag: imageName,
+        child: new RadialExpansion(
+          maxRadius: kMaxRadius,
+          child: new Photo(
+            photo: imageName,
+            onTap: () {
+              Navigator.of(context).push(
+                new PageRouteBuilder<Null>(
+                  pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+                    return new AnimatedBuilder(
+                      animation: animation,
+                      builder: (BuildContext context, Widget child) {
+                        return new Opacity(
+                          opacity: opacityCurve.transform(animation.value),
+                          child: _buildPage(context, imageName, description),
+                        );
+                      }
+                    );
+                  },
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    timeDilation = 5.0; // 1.0 is normal animation speed.
+
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Radial Transition Demo'),
+      ),
+      body: new Container(
+        padding: const EdgeInsets.all(32.0),
+        alignment: FractionalOffset.bottomLeft,
+        child: new Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            _buildHero(context, 'images/chair-alpha.png', 'Chair'),
+            _buildHero(context, 'images/binoculars-alpha.png', 'Binoculars'),
+            _buildHero(context, 'images/beachball-alpha.png', 'Beach ball'),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+void main() {
+  runApp(new MaterialApp(home: new RadialExpansionDemo()));
+}
```

Comment:
When I ran this code, I couldn't load the three images. I copied the assets and the pubspec.yaml file. I was getting these errors:

```
[VERBOSE-2:dart_error.cc(16)] Unhandled exception:
	Unable to load asset: AssetManifest.json
	#0      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:221:7)
	<asynchronous suspension>
	#1      CachingAssetBundle._fetchString (package:flutter/src/services/asset_bundle.dart:160:33)
	<asynchronous suspension>
	#2      CachingAssetBundle.loadString (package:flutter/src/services/asset_bundle.dart:156:12)
	#3      CachingAssetBundle.loadStructuredData (package:flutter/src/services/asset_bundle.dart:184:5)
	#4      AssetImage.obtainKey (package:flutter/src/services/image_resolution.dart:167:18)
	#5      ImageProvider.resolve (package:flutter/src/services/image_provider.dart:264:5)
	#6      _ImageState._resolveImage (package:flutter/src/widgets/image.dart:468:33)
	#7      _ImageState.didChangeDependencies (package:flutter/src/widgets/image.dart:449:5)
	#8      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3666:12)
	#9      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3525:5)
	#10     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
	#11     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
	#12     _LayoutBuilderElement._layout.<anonymous closure> (package:flutter/src/widgets/layout_builder.dart:119:18)
	#13     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2231:19)
	#14     _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:107:11)
	#15     RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2058:58)
	#16     PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:1148:15)
	#17     RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2058:13)
	#18     _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:207:5)
	#19     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#20     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#21     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#22     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#23     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#24     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#25     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#26     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#27     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1005:24)
	#28     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#29     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#30     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1005:24)
	#31     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#32     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:257:13)
	#33     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#34     RenderPositionedBox.performLayout (package:flutter/src/rendering/shifted_box.dart:373:13)
	#35     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#36     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#37     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1005:24)
	#38     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#39     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:257:13)
	#40     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#41     RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:737:15)
	#42     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#43     RenderPositionedBox.performLayout (package:flutter/src/rendering/shifted_box.dart:373:13)
	#44     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#45     RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:197:11)
	#46     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#47     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:124:11)
	#48     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:91:7)
	#49     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:194:7)
	#50     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:338:14)
	#51     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#52     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#53     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#54     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#55     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1005:24)
	#56     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#57     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#58     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#59     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:466:15)
	#60     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#61     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#62     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#63     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#64     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#65     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#66     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#67     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#68     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#69     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#70     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:2749:26)
	#71     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#72     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#73     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#74     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:466:15)
	#75     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#76     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#77     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#78     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#79     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#80     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#81     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#82     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#83     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#84     RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:107:13)
	#85     RenderObject.layout (package:flutter/src/rendering/object.dart:1962:7)
	#86     RenderView.performLayout (package:flutter/src/rendering/view.dart:142:13)
	#87     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1837:7)
	#88     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1126:18)
	#89     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:262:19)
	#90     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:581:22)
	#91     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:200:5)
	#92     BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:713:15)
	#93     BindingBase&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:649:9)
	#94     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/rendering/binding.dart:286:20)
	#95     Timer._createTimer.<anonymous closure> (dart:async-patch/dart:async/timer_patch.dart:21)
	#96     _Timer._runTimers (dart:isolate-patch/dart:isolate/timer_impl.dart:367)
	#97     _Timer._handleMessage (dart:isolate-patch/dart:isolate/timer_impl.dart:401)
	#98     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:163)
```

Flutter Doctor output:
```
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel master)
    • Flutter at /Users/taodong/flutter
    • Framework revision 17db2d2335 (2 hours ago), 2017-10-13 16:38:24 -0700
    • Engine revision 248af2d5b6
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /Users/taodong/Library/Android/sdk
    • Platform android-25, build-tools 25.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.2
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2.3)
    • Flutter plugin version 18.1
    • Dart plugin version 172.3757.55

[✓] Connected devices
    • iPhone 6s • B532439D-E9C6-4B65-A6C4-44388EFC0B8E • ios • iOS 10.3 (simulator)
```

Reply:
I tried the code on an Android physical device, and it was fine. So this has something to do with the iOS Simulator.

Reply:
I have seen that in the Simulator, but I find if I try two or three times, it works fine. If you can reproduce it, can you file a bug against Simulator?

Reply:
This has been fixed in the latest master.